routes/api/vote.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import type { Handlers, PageProps } from "$fresh/server.ts";
+import type { State } from "@/routes/_middleware.ts";
+import { createVote, deleteVote } from "@/utils/db.ts";
+
+export const handler: Handlers<PageProps, State> = {
+  async POST(req, ctx) {
+    if (!ctx.state.session) {
+      return new Response(null, { status: 400 });
+    }
+
+    const params = new URL(req.url).searchParams;
+    const itemId = params.get("to");
+
+    if (itemId) {
+      await createVote({
+        userId: ctx.state.session.user.id,
+        itemId,
+      });
+    }
```

Comment:
We should first guard against if `itemId` is `null` by returning a `new Response(null, { status: 400 })`.